source/uwp/Renderer/lib/MediaHelpers.cpp

```diff
@@ -215,7 +242,30 @@ namespace AdaptiveCards::Rendering::Uwp::MediaHelpers
 
             if (resourceResolver == nullptr)
             {
-                mediaElement.Source(mediaSourceUrl);
+                auto mediaSrc = winrt::MediaSource::CreateFromUri(mediaSourceUrl);
+                if (adaptiveMedia.CaptionSources().Size() > 0)
+                {
+                    for (const auto captionSource : adaptiveMedia.CaptionSources())
+                    {
+                        const auto timedTextURL = GetUrlFromString(renderContext.HostConfig(), captionSource.Url());
+                        const auto timedTextSrc = winrt::TimedTextSource::CreateFromUri(timedTextURL);
+                        timedTextSrc.Resolved([captionSource](winrt::TimedTextSource sender,
+                                                              winrt::TimedTextSourceResolveResultEventArgs args) {
+                                if (!args.Error()){
+                                    args.Tracks().GetAt(0).Label(captionSource.MimeType());
+                                }
+                            });
+                        mediaSrc.ExternalTimedTextSources().Append(timedTextSrc);
+                    }
+                }
+                auto playbackItem = winrt::MediaPlaybackItem(mediaSrc);
+                playbackItem.TimedMetadataTracksChanged(
+                    [playbackItem](winrt::IInspectable const& /*sender*/, winrt::IInspectable const& /*args*/)
+                    {
+                        playbackItem.TimedMetadataTracks().SetPresentationMode(
+                            0, winrt::TimedMetadataTrackPresentationMode::PlatformPresented);
+                    });
+                mediaElement.Source(playbackItem);
```

Comment:
Could you please extract that code into a function to not repeat the code twice?:)